x-pack/plugins/security_solution_serverless/server/task_manager/usage_reporting_task.test.ts

```diff
@@ -83,60 +89,62 @@ describe('SecurityUsageReportingTask', () => {
     };
   }
 
-  function buildTaskArgs({
-    core,
-    taskManager,
-    meteringCallback,
-  }: {
-    core: CoreSetup;
-    taskManager: TaskManagerSetupContract;
-    meteringCallback: MeteringCallback;
-  }): SecurityUsageReportingTaskSetupContract {
-    return {
-      core,
-      logFactory: loggingSystemMock.create(),
-      config: {
-        productTypes: [
-          {
-            product_line: ProductLine.security,
-            product_tier: ProductTier.complete,
+  function buildTaskArgs(
+    overrides?: Partial<SecurityUsageReportingTaskSetupContract>
+  ): SecurityUsageReportingTaskSetupContract {
+    return assign(
+      {
+        core: mockCore,
+        logFactory: loggingSystemMock.create(),
+        config: {
+          productTypes: [
+            {
+              product_line: ProductLine.security,
+              product_tier: ProductTier.complete,
+            },
+          ],
+        } as ServerlessSecurityConfig,
+        taskManager: mockTaskManagerSetup,
+        cloudSetup: {
+          serverless: {
+            projectId: PROJECT_ID,
           },
-        ],
-      } as ServerlessSecurityConfig,
-      taskManager,
-      cloudSetup: {
-        serverless: {
-          projectId: PROJECT_ID,
-        },
-      } as CloudSetup,
-      taskType: TYPE,
-      taskTitle: TITLE,
-      version: VERSION,
-      meteringCallback,
-    };
+        } as CloudSetup,
+        taskType: TYPE,
+        taskTitle: TITLE,
+        version: VERSION,
+        meteringCallback: meteringCallbackMock,
+      },
+      overrides
+    );
   }
 
-  beforeEach(async () => {
+  async function setupMocks() {
     mockCore = coreSetupMock();
     mockEsClient = (await mockCore.getStartServices())[0].elasticsearch.client
       .asInternalUser as jest.Mocked<ElasticsearchClient>;
     mockTaskManagerSetup = tmSetupMock();
     usageRecord = buildUsageRecord();
+    reportUsageSpy = jest.spyOn(usageReportingService, 'reportUsage');
     meteringCallbackMock = jest.fn().mockResolvedValueOnce([usageRecord]);
-    taskArgs = buildTaskArgs({
-      core: mockCore,
-      taskManager: mockTaskManagerSetup,
-      meteringCallback: meteringCallbackMock,
-    });
+    taskArgs = buildTaskArgs();
     mockTask = new SecurityUsageReportingTask(taskArgs);
+  }
+
+  beforeEach(async () => {
+    await setupMocks();
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
   });
 
   describe('task lifecycle', () => {
-    it('should create task', () => {
+    it('should create task', async () => {
       expect(mockTask).toBeInstanceOf(SecurityUsageReportingTask);
     });
 
-    it('should register task', () => {
+    it('should register task', async () => {
```

Comment:
Q: do these need to be `async`?

Reply:
nope, definitely forgot to change these back. good catch 👍.